components/fxa-client/src/fxa_client.udl

```diff
@@ -138,6 +138,11 @@ interface FirefoxAccount {
   [Throws=FxaError]
   string to_json();
   
+  // Sets the users information based on the web content's login information
+  // This is intended to only be used by user agents (eg: Firefox) to set the users
+  // session token and tie it to the refresh token that will be issued at the end of the
+  // oauth flow.
+  void set_user_data(UserData user_data);
```

Comment:
I'm not sure, but I kind of think this data should be sent via a state machine event.  Could it be an optional field on `FxaEvent::CompleteOAuthFlow`?  Alternatively, could there be a `FxaEvent::StoreUserData` event?

Reply:
We chatted a bit about this and decided that it would be a nice followup but not necessary here - resolving!